### PR TITLE
add characteristic_polynomial

### PR DESCRIPTION
**Pull request by hly1204**
<https://github.com/yosupo06/library-checker-problems/issues/665>

Thanks!

### CODE REVIEW

File: math/characteristic_polynomial/task.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+## @{keyword.statement}
+
+@{lang.en}
+
+Given $N \times N$ matrix $M = \lbrace a_{ij} \rbrace$. Print $p(x)$ where
+
+$$p(x) = \det (xI - M) = \sum_{i = 0}^N p_i x^i$$
+
+and $I$ denotes the $N \times N$ identity matrix. We consider everything $\mathbb{Z}_{@{param.MOD}}$.
```
Comment (hos-lyric) [inline math/characteristic_polynomial/task.md:9]: Maybe "in $\mathbb{Z}/@{param.MOD}\mathbb{Z}$" sounds better?

File: math/characteristic_polynomial/task.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+$$p(x) = \det (xI - M) = \sum_{i = 0}^N p_i x^i$$
+
+ここで、$I$ は $N \times N$ の単位行列です。全ての要素は $\mathbb{Z}_{@{param.MOD}}$ で考えることとします。
+
+@{lang.end}
+
+## @{keyword.constraints}
+
+- $2 \leq N \leq @{param.N_MAX}$
```
Comment (hos-lyric) [inline math/characteristic_polynomial/task.md:23]: Sorry that I didn't mention when proposed, but I think we can include N = 1 (and possibly N = 0) as well.

**Review by hos-lyric — COMMENTED** [review]
Thanks!

**Comment by hly1204** [comment]
> Thanks!

Thanks for reviewing!
I will fix all these problems. I didn't consider the corner case before, thanks for pointing out!
About the N = 0 case, I will give an explicit definition as same as the paper showed. ($p(x) = 1$ when N = 0)

**Comment by hos-lyric** [comment]
LGTM

**Comment by hitonanode** [comment]
It seems that the testcases are a little weak. The code that outputs not the characteristic but minimal polynomials can pass almost all testcases.
https://judge.yosupo.jp/submission/50226
This wrong submission can be hacked by the inputs such as:
```
4
1 0 0 0
0 1 0 0
0 0 1 0
0 0 0 1
```

**Pull request merged**